azurerm/data_source_batch_pool.go

```diff
@@ -0,0 +1,159 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchPool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMBatchPoolName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vm_size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"scale_mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_dedicated_nodes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"target_low_priority_nodes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"resize_timeout": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"autoscale_evaluation_interval": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"autoscale_formula": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_image_reference": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"publisher": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"version": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
Should the ForceNew and optional be removed?

Reply:
Yep it should. Done.